### PR TITLE
Refactor - SQLRepository uses sqlite3 instead of sqlalchemy

### PR DESCRIPTION
**Pull request by toan-quach**

### CODE REVIEW

File: src/taipy/core/_repository/_sql_repository.py [hunk 1]
```diff
@@ -35,8 +36,7 @@ def __init__(self, model_type: Type[ModelType], converter: Type[Converter], sess
             converter: A class that handles conversion to and from a database backend
             db: An SQLAlchemy session object
```
Comment (trgiangdo) [inline src/taipy/core/_repository/_sql_repository.py]:
```suggestion
            db: An sqlite3 connection object
```

File: src/taipy/core/_repository/_sql_repository.py [hunk 2]
```diff
@@ -45,56 +45,79 @@ def __init__(self, model_type: Type[ModelType], converter: Type[Converter], sess
...
-            raise ModelNotFound(str(self.model_type.__name__), entity_id)
+        delete_query = self.model_type.__table__.delete().filter_by(id=entity_id)  # type: ignore
+        cursor = self.db.execute(str(delete_query.compile(dialect=sqlite.dialect())), [entity_id])
         self.db.commit()
 
+        if cursor.rowcount == 0:
+            raise ModelNotFound(str(self.model_type.__name__), entity_id)
+
+        if cursor.rowcount == 0:
+            raise ModelNotFound(str(self.model_type.__name__), entity_id)
```
Comment (trgiangdo) [inline src/taipy/core/_repository/_sql_repository.py]: The `if` block is duplicated I believe. Also, should we run the `if` before commit the connection?

```suggestion
        if cursor.rowcount == 0:
            raise ModelNotFound(str(self.model_type.__name__), entity_id)
            
        self.db.commit()
```

**Review by trgiangdo — COMMENTED** [review]
I have some small comments.

Other than that, there are a lot of `# type: ignore` due to the `ModelType` in _sql_repository. We will need to come back to the typing in the future.

**Review by trgiangdo — DISMISSED** [review]
A lot of `# type: ignore` have been removed. Neat :D

**Comment by florian-vuillemot** [comment]
Can "sqlachemy" be removed from setup.py and Pipfile?

**Comment by toan-quach** [comment]
@florian-vuillemot sadly no since we still need it for the SQL Datanode

File: src/taipy/core/_repository/_encoder.py [hunk 3]
```diff
@@ -27,7 +25,7 @@ def _timedelta_to_str(self, obj: timedelta) -> str:
             f"{int(total_seconds % 60)}s"
         )
 
-    def default(self, o: Any) -> Json:
```
Comment (jrobinAV) [inline src/taipy/core/_repository/_encoder.py:30]: Why do we need to remove the typing?

File: src/taipy/core/job/_job_model.py [hunk 4]
```diff
@@ -66,8 +58,8 @@ def from_dict(data: Dict[str, Any]):
             submit_id=data["submit_id"],
             submit_entity_id=data["submit_entity_id"],
             creation_date=data["creation_date"],
-            subscribers=subscribers,
-            stacktrace=stacktrace,
+            subscribers=_BaseModel._deserialize_attribute(data["subscribers"]),
+            stacktrace=data["stacktrace"],
```
Comment (jrobinAV) [inline src/taipy/core/job/_job_model.py]: don't we want to use the deserialize method?

File: src/taipy/core/_repository/_base_taipy_model.py [hunk 5]
```diff
@@ -26,3 +30,13 @@ def to_dict(self) -> Dict[str, Any]:
...
+
+    @staticmethod
+    def _serialize_attribute(value):
+        return json.dumps(value, ensure_ascii=False, cls=_Encoder)
+
+    @staticmethod
+    def _deserialize_attribute(value):
+        if isinstance(value, str):
+            return json.loads(value.replace("'", '"'), cls=_Decoder)
+        return value
```
Comment (jrobinAV) [inline src/taipy/core/_repository/_base_taipy_model.py:46]: Much better with serialize and deserialize methods.

Thanks!!

File: src/taipy/core/_version/_version_model.py [hunk 6]
```diff
@@ -37,8 +37,22 @@ class _VersionModel(_BaseModel):
...
+        return model
+
+    def to_list(self):
+        return [
+            self.id,
+            self.config,
+            self.creation_date,
+            self.is_production,
+            self.is_development,
+            self.is_latest,
```
Comment (jrobinAV) [inline src/taipy/core/_version/_version_model.py:57]: If I am not mistaken, the order of this list must be consistent with the order of the columns.  Am I correct?
 Do we have a way to test our code to ensure that?

**Comment by jrobinAV** [comment]
> I have some small comments.
> 
> Other than that, there are a lot of `# type: ignore` due to the `ModelType` in _sql_repository. We will need to come back to the typing in the future.

Why not now? Do you think it will block the PR for too long? or do you believe it will make the PR too messy?

File: src/taipy/core/job/_job_model.py (hunk 4 above)
Comment (jrobinAV) [reply src/taipy/core/job/_job_model.py]: If yes, that means we are missing a unit test abt the stacktrace attribute.

File: src/taipy/core/_version/_version_model.py (hunk 6 above)
Comment (toan-quach) [reply src/taipy/core/_version/_version_model.py:57]: ah yes, the order of list must be consistent with the column orders we defined in __table__

File: src/taipy/core/_repository/_encoder.py (hunk 3 above)
Comment (toan-quach) [reply src/taipy/core/_repository/_encoder.py:30]: it's due to a circular import issue

File: src/taipy/core/job/_job_model.py (hunk 4 above)
Comment (toan-quach) [reply src/taipy/core/job/_job_model.py]: I think you're right, it raised an error when I attempt to use deserialize here, will have a look

File: src/taipy/core/_repository/_encoder.py (hunk 3 above)
Comment (jrobinAV) [reply src/taipy/core/_repository/_encoder.py:30]: ok thx

File: src/taipy/core/job/_job_model.py (hunk 4 above)
Comment (toan-quach) [reply src/taipy/core/job/_job_model.py]: yep it worked now, I think there was some issue with multithreading but got fixed. Though will need unit tests

**Review by trgiangdo — APPROVED** [review]

**Review by jrobinAV — APPROVED** [review]

**Review by joaoandre-avaiga — APPROVED** [review]

**Pull request merged**